dune.opam.template

```diff
@@ -1,6 +1,4 @@
 build: [
-  # opam 2 sets OPAM_SWITCH_PREFIX, so we don't need a hardcoded path
-  ["ocaml" "configure.ml" "--libdir" lib] {opam-version < "2"}
```

Comment:
This is the only part of the PR that's an opam 2 special case, right? The rest is about allowing a more flexible default for the lib dirs. I don't see what that has to do with opam 2?